File: src/Chapter05/SlickUpload-5.5.1/Samples-1.1/SamplesVB/FileNameGenerator/PrefixFolderFileNameGenerator.vb
Imports System.IO

Imports Krystalware.SlickUpload
Imports Krystalware.SlickUpload.Providers

Namespace FileNameGenerator
    ''' <summary>
    ''' An <see cref="IFileNameGenerator" /> that puts each file into a folder named for the first character
    ''' in the file naMyBase.
    ''' </summary>
    Public Class PrefixFolderFileNameGenerator
        Implements IFileNameGenerator

        Public Function GenerateFileName(ByVal file As Krystalware.SlickUpload.UploadedFile) As String Implements Krystalware.SlickUpload.Providers.IFileNameGenerator.GenerateFileName
            Return Path.Combine(file.ClientName.Substring(0, 1), file.ClientName)
        End Function
    End Class
End Namespace

File: src/Chapter05/SlickUpload-5.5.1/Samples-1.1/SamplesCS/FileNameGenerator/PrefixFolderFileNameGenerator.cs
﻿using System;
using System.IO;

using Krystalware.SlickUpload;
using Krystalware.SlickUpload.Providers;

namespace SamplesCS.FileNameGenerator
{
	/// <summary>
	/// An <see cref="IFileNameGenerator" /> that puts each file into a folder named for the first character
	/// in the file name.
	/// </summary>
	public class PrefixFolderFileNameGenerator : IFileNameGenerator
	{
		public string GenerateFileName(UploadedFile file)
		{
			return Path.Combine(file.ClientName.Substring(0, 1), file.ClientName);
		}
	}
}